Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -954,20 +954,21 @@ int RenderBox::horizontalScrollbarHeight() const
     return includeHorizontalScrollbarSize() ? scrollableArea->horizontalScrollbarHeight(IgnoreOverlayScrollbarSize, isHorizontalWritingMode()) : 0;
 }
 
+static bool shouldShowScrollbarGutter(ScrollbarGutter gutter, bool hasVisibleOverflow, Overflow overflow)
+{
+    return overflow == Overflow::Scroll || (overflow == Overflow::Auto && (!gutter.isAuto || hasVisibleOverflow)) || (overflow == Overflow::Hidden && !gutter.isAuto);
+}
+
 int RenderBox::intrinsicScrollbarLogicalWidth() const
```

Comment:
It's not quite right to sometimes return ` verticalScrollbarWidth() * 2` from a function that purportedly returns the scrollbar width; it would be easy to write code that assumes that the return value is actually for the scrollbar. Either rename this or put the gutter code somewhere else.

Reply:
Changed the name to `intrinsicScrollbarLogicalWidthIncludingGutter`.